src/PProf.jl

```diff
@@ -417,9 +418,32 @@ end
 
 # Precompile as much as possible, so that profiling doesn't end up measuring our own
 # compilation.
-function __init__()
-    precompile(pprof, ()) || error("precompilation of package functions is not supposed to fail")
-    precompile(kill, ()) || error("precompilation of package functions is not supposed to fail")
-    precompile(refresh, ()) || error("precompilation of package functions is not supposed to fail")
+@static if Base.VERSION >= v"1.8.0-"
+    @setup_workload begin
+        using Profile
+        f1() = sum(collect(1:1e7))
+        f1()
+        f2() = [[] for _ in 1:5]
+        f2()
```

Comment:
i presume we call these here in `@setup_workload` to avoid `precompile(f2, ())` being part of the resulting package

Reply:
Yep, exactly